core/src/main/scala/chisel3/Bits.scala

```diff
@@ -676,8 +677,26 @@ sealed class UInt private[chisel3] (width: Width) extends Bits(width) with Num[U
     this << castToInt(that, "Shift amount")
   override def do_<<(that: UInt)(implicit sourceInfo: SourceInfo): UInt =
     binop(sourceInfo, UInt(this.width.dynamicShiftLeft(that.width)), DynamicShiftLeftOp, that)
-  override def do_>>(that: Int)(implicit sourceInfo: SourceInfo): UInt =
-    binop(sourceInfo, UInt(this.width.shiftRight(that)), ShiftRightOp, validateShiftAmount(that))
+
+  // Implement legacy [buggy] UInt shr behavior for both Chisel and FIRRTL
+  @nowarn("msg=method shiftRight in class Width is deprecated")
```

Comment:
What does nowarn mean again?

Reply:
I deprecated `Width.shiftRight` and it would trigger a warning here (and warnings are errors), so this `@nowarn` suppresses it. Obviously we are allowed to use our own deprecated APIs, but the API is public so I couldn't/shouldn't just delete it.